packages/core/core-flows/src/cart/steps/set-tax-lines-for-items.ts

```diff
@@ -35,18 +35,15 @@ export const setTaxLinesForItemsStep = createStep(
     })
 
     const itemsTaxLinesData = normalizeItemTaxLinesForCart(item_tax_lines)
-    const setItemTaxLinesPromise = itemsTaxLinesData.length
-      ? cartService.setLineItemTaxLines(cart.id, itemsTaxLinesData)
-      : 0
+    const setItemTaxLinesPromise = cartService.setLineItemTaxLines(
+      cart.id,
+      itemsTaxLinesData
+    )
 
     const shippingTaxLinesData =
       normalizeShippingTaxLinesForCart(shipping_tax_lines)
-    const setShippingTaxLinesPromise = shippingTaxLinesData.length
-      ? await cartService.setShippingMethodTaxLines(
-          cart.id,
-          shippingTaxLinesData
-        )
-      : 0
+    const setShippingTaxLinesPromise =
+      await cartService.setShippingMethodTaxLines(cart.id, shippingTaxLinesData)
```

Comment:
In the case of no tax region for the provided country, `shippingTaxLinesData` and `itemsTaxLinesData` will be empty arrays and that should be reflected on the cart.